Source/WebCore/platform/network/NetworkStorageSession.h

```diff
@@ -157,20 +158,27 @@ class NetworkStorageSession
     WEBCORE_EXPORT ~NetworkStorageSession();
 #endif
 
+#if PLATFORM(COCOA) || USE(SOUP)
+    enum class IsInMemoryCookieStore : bool { No, Yes };
+#endif
+
 #if PLATFORM(COCOA)
     enum class ShouldDisableCFURLCache : bool { No, Yes };
     WEBCORE_EXPORT static RetainPtr<CFURLStorageSessionRef> createCFStorageSessionForIdentifier(CFStringRef identifier, ShouldDisableCFURLCache = ShouldDisableCFURLCache::No);
-    enum class IsInMemoryCookieStore : bool { No, Yes };
     WEBCORE_EXPORT NetworkStorageSession(PAL::SessionID, RetainPtr<CFURLStorageSessionRef>&&, RetainPtr<CFHTTPCookieStorageRef>&&, IsInMemoryCookieStore = IsInMemoryCookieStore::No);
     WEBCORE_EXPORT explicit NetworkStorageSession(PAL::SessionID);
 
     // May be null, in which case a Foundation default should be used.
     CFURLStorageSessionRef platformSession() { return m_platformSession.get(); }
     WEBCORE_EXPORT RetainPtr<CFHTTPCookieStorageRef> cookieStorage() const;
 #elif USE(SOUP)
-    WEBCORE_EXPORT explicit NetworkStorageSession(PAL::SessionID);
+    WEBCORE_EXPORT explicit NetworkStorageSession(PAL::SessionID, IsInMemoryCookieStore = IsInMemoryCookieStore::No);
     ~NetworkStorageSession();
 
+    enum class ForHTTPHeader : bool { No, Yes };
+    std::optional<GSList*> lookupCookies(const URL& firstParty, const SameSiteInfo&, const URL&, ForHTTPHeader, std::optional<FrameIdentifier>, std::optional<PageIdentifier>, ApplyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking, IncludeSecureCookies, bool* didAccessSecureCookies = nullptr) const;
+    std::pair<String, bool> lookupCookiesHeaders(const URL& firstParty, const SameSiteInfo&, const URL&, ForHTTPHeader, std::optional<FrameIdentifier>, std::optional<PageIdentifier>, ApplyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking, IncludeSecureCookies) const;
```

Comment:
The argument list,other than `didAccessSecureCookies`, is shared between these two methods, and it's lengthy enough that maybe we could consider extracting them into a helper structure?

Reply:
I did this originally but it didn't really do anything of value IMO.

Like the function signature is more simple `std::pair<String, bool> lookupCookiesHeaders(const CookieLookupData&);`

But the usage is *identical*:

```cpp
lookupCookiesHeaders({
        firstParty,
        sameSiteInfo,
        url,
        ForHTTPHeader::Yes,
        frameID,
        pageID,
        applyTrackingPrevention,
        relaxThirdPartyCookieBlocking,
        includeSecureCookies});
```

So I decided to avoid defining a custom type in this header since its shared between all ports, so simple is better.

Dunno if anyone else has an opinion on it.

Reply:
I wouldn't use a helper structure for this.